packages/flow-designer/src/components/FlowDesigner.container.tsx

```diff
@@ -39,63 +39,32 @@ type Props = {
 	snapToGrid?: boolean;
 };
 
+type StateMap = { [key: string]: any };
+
 type State = {
-	nodeTypeMap: Map<string, any>;
-	linkTypeMap: Map<string, any>;
-	portTypeMap: Map<string, any>;
+	nodeTypeMap: StateMap;
+	linkTypeMap: StateMap;
+	portTypeMap: StateMap;
 };
 
-export class FlowDesigner extends React.Component<Props, State> {
-	node: any;
-
-	static defaultProps = {
-		snapToGrid: false,
-	};
-
-	constructor(props: Props) {
-		super(props);
-		this.state = {
-			nodeTypeMap: Map(),
-			linkTypeMap: Map(),
-			portTypeMap: Map(),
-		};
-	}
-
-	UNSAFE_componentWillMount() {
```

Comment:
This had 2 objectives: init state at mount and call props.setNodeTypes at mount
Split it into 
- useState init function to init the state
- useEffect once to call props.setNodeTypes